cloudbuild.yaml

```diff
@@ -0,0 +1,111 @@
+steps:
+
+# Create docker image from regular Dockerfile
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['build', '--tag=gcr.io/google.com/hadoop-cloud-dev/init-actions-image', '.']
+# Run tests from the repo, based on the branch name
```

Comment:
Instead of running all test lets run tests only for modified init actions. We may need to create shell or python script that does this and execute it here.

Reply:
In here, only certain init actions are run (based on the branch name).

I propose changing this approach to having several defined set of build steps (different yaml files), each representing a given feature to be tested and triggered only when their respective files are modified. The downside is that we'll have many yaml files (one for each feature).

Reply:
This functionality is supported by GitHub App Triggers Alpha and we would end up with multiple triggers defined in GCP.

Reply:
I think that this is nor reliable, because this convention is not obvious and requires human attention to enforce branch names.

What we need to do is to have script that diffs PR branch against master branch and after that runs tests for all init actions that modified by PR. Is it feasible to implement this with Cloud Build?

Reply:
Yes, I've been working on this approach. I can run a different set of steps based on files modified. CB GitHub App Triggers supports this by triggering builds depending on files included in a given PR. I'll have to define each trigger in GCP.

Reply:
Is CB GitHub App Triggers allow to configure this in generic way? i.e if `init-action-folder` is modified run `init-action-folder/test.py`? I'm asking, because I don't want to modify them each time we are adding new init action.

Reply:
Yes, it allows us to define a generic path (e.g. when bigtable/** changes, we run bigtable/test_bigtable.py). If no changes are made to the files in the init action folder, its test is not triggered. For each init action, a generic rule like this will need to be defined.

Reply:
I guess this is not generic enough comparing to what I thought, because it requires per-folder/init action pattern definition.